RecoLocalTracker/SiStripClusterizer/interface/ThreeThresholdAlgorithm.h

```diff
@@ -9,70 +9,66 @@ class ThreeThresholdAlgorithm final : public StripClusterizerAlgorithm {
 
-  bool stripByStripBegin(uint32_t id);
+  void clusterizeDetUnit(const    edm::DetSet<SiStripDigi> &, output_t::FastFiller &) const;
+  void clusterizeDetUnit(const edmNew::DetSet<SiStripDigi> &, output_t::FastFiller &) const;
+
+  Det stripByStripBegin(uint32_t id) const;
 
   // LazyGetter interface
-  void stripByStripAdd(uint16_t strip, uint8_t adc, std::vector<SiStripCluster>& out);
-  void stripByStripEnd(std::vector<SiStripCluster>& out);
-  void addFed(sistrip::FEDZSChannelUnpacker & unpacker, uint16_t ipair, std::vector<SiStripCluster>& out) {
+  void stripByStripAdd(State & state, uint16_t strip, uint8_t adc, std::vector<SiStripCluster>& out) const;
+  void stripByStripEnd(State & state, std::vector<SiStripCluster>& out) const;
+
+  void addFed(State & state, sistrip::FEDZSChannelUnpacker & unpacker, uint16_t ipair, std::vector<SiStripCluster>& out) const {
     while (unpacker.hasData()) {
-      stripByStripAdd(unpacker.sampleNumber()+ipair*256,unpacker.adc(),out);
+      stripByStripAdd(state,unpacker.sampleNumber()+ipair*256,unpacker.adc(),out);
       unpacker++;
     }
   }
 
   // detset interface
-  void addFed(sistrip::FEDZSChannelUnpacker & unpacker, uint16_t ipair, output_t::FastFiller & out) override {
+  void addFed(State & state, sistrip::FEDZSChannelUnpacker & unpacker, uint16_t ipair, output_t::FastFiller & out) const override {
     while (unpacker.hasData()) {
-      stripByStripAdd(unpacker.sampleNumber()+ipair*256,unpacker.adc(),out);
+      stripByStripAdd(state, unpacker.sampleNumber()+ipair*256,unpacker.adc(),out);
       unpacker++;
     }
   }
 
-  void stripByStripAdd(uint16_t strip, uint8_t adc, output_t::FastFiller & out) override {
-    if(candidateEnded(strip)) endCandidate(out);
-    addToCandidate(strip,adc);
+  void stripByStripAdd(State & state, uint16_t strip, uint8_t adc, output_t::FastFiller & out) const override {
+    if(candidateEnded(state, strip)) endCandidate(state, out);
+    addToCandidate(state, strip,adc);
   }
 
-  void stripByStripEnd(output_t::FastFiller & out) override { endCandidate(out);}
-
-  void cleanState() override {clearCandidate();}
+  void stripByStripEnd(State & state, output_t::FastFiller & out) const override { endCandidate(state,out);}
 
 
  private:
 
-  template<class T> void clusterizeDetUnit_(const T&, output_t::FastFiller&);
+  template<class T> void clusterizeDetUnit_(const T&, output_t::FastFiller&) const;
   ThreeThresholdAlgorithm(float, float, float, unsigned, unsigned, unsigned, std::string qualityLabel,
 			  bool setDetId, bool removeApvShots, float minGoodCharge);
 
-  //state of the candidate cluster
-  std::vector<uint8_t> ADCs;  
-  uint16_t lastStrip;
-  float noiseSquared;
-  bool candidateLacksSeed;
 
-  //constant methods with state information
-  uint16_t firstStrip() const {return lastStrip - ADCs.size() + 1;}
-  bool candidateEnded(const uint16_t&) const;
-  bool candidateAccepted() const;
+    //constant methods with state information
```

Comment:
github web diff suggests there is a spurious change in indentation for the modified methods (until L65).
Better keep it consistent with the rest